sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosDiagnosticsE2ETest.java

```diff
@@ -177,10 +204,46 @@ public void onlyCustomLoggerWithCustomConfig() {
                     .diagnosticsHandler(capturingLogger)
             );
         CosmosContainer container = this.getContainer(builder);
-        executeTestCase(container);
+        String id = UUID.randomUUID().toString();
+        ObjectNode doc = getDocumentDefinition(id);
+
+        CosmosDiagnostics diagnostics = executeDocumentOperation(container, OperationType.Create, id, doc);
+        assertThat(diagnostics).isNotNull();
+        assertThat(diagnostics.getDiagnosticsContext()).isNotNull();
+
+        diagnostics = executeDocumentOperation(container, OperationType.Query, id, doc);
+        assertThat(diagnostics).isNotNull();
+        assertThat(diagnostics.getDiagnosticsContext()).isNotNull();
 
         assertThat(capturingLogger.getLoggedMessages()).isNotNull();
-        assertThat(capturingLogger.getLoggedMessages()).hasSize(1);
+        assertThat(capturingLogger.getLoggedMessages()).hasSize(2);
+    }
+
+    @Test(groups = { "fast", "emulator" }, timeOut = TIMEOUT)
+    public void onlyCustomLoggerAlwaysSampledOut() {
+
+        CapturingLogger capturingLogger = new CapturingLogger();
+
+        CosmosClientBuilder builder = this
+            .getClientBuilder()
+            .clientTelemetryConfig(
+                new CosmosClientTelemetryConfig()
+                    .diagnosticsThresholds(
+                        new CosmosDiagnosticsThresholds()
+                            .setPointOperationLatencyThreshold(Duration.ofMillis(100))
+                            .setNonPointOperationLatencyThreshold(Duration.ofMillis(2000))
+                            .setRequestChargeThreshold(100)
+                    )
+                    .diagnosticsHandler(capturingLogger)
+                    .sampleDiagnostics(0)
```

Comment:
Can we also add a test case where we actually sampleDiagnostics with value > 0 and < 1?

Reply:
what purpose would that have? That test would be flaky by design - unless you use a high enough iteration count to force hitting each of the two code paths being hit?

Reply:
Discussed offline - validation has been added for sampling rate 1 and 0 - so all code paths are covered by the tests added.